test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -3314,6 +3314,35 @@ BOOST_AUTO_TEST_CASE(using_enums)
 	BOOST_CHECK(callContractFunction("getChoice()") == encodeArgs(2));
 }
 
+BOOST_AUTO_TEST_CASE(enum_explicit_overflow)
+{
+	char const* sourceCode = R"(
+			contract test {
+				enum ActionChoices { GoLeft, GoRight, GoStraight }
+				enum Dummy { Dummy0, Dummy1, Dummy2, Dummy3 }
+				function test()
+				{
+				}
+				function getChoiceExp() returns (uint d)
+				{
+					choices = ActionChoices(3);
+					d = uint256(choices);
+				}
+				function getChoiceFromEnum() returns (uint d)
+				{
+					choices = ActionChoices(Dummy.Dummy3);
+					d = uint256(choices);
+				}
+				ActionChoices choices;
+			}
+	)";
+	compileAndRun(sourceCode);
+	// throws
+	BOOST_CHECK(callContractFunction("getChoiceExp()") == encodeArgs());
+	// also throws
+	BOOST_CHECK(callContractFunction("getChoiceFromEnum()") == encodeArgs());
```

Comment:
Can you add tests for the edge case (value == limit)?

Do we have also have tests for the other success cases?